lib/gooddata/lcm/actions/associate_clients.rb

```diff
@@ -46,7 +49,21 @@ class AssociateClients < BaseAction
       ]
 
       class << self
+        DELETE_MODES = %w(
+          none
+          delete_projects
+          delete_extra
+        )
         def call(params)
+          if params.delete_projects && !params.delete_extra
+            GoodData.logger.warn("Parameter `delete_projects` is set to true and parameter `delete_extra` is set to false (which is default) this action will do nothing!")
```

Comment:
If the combination is not supported, shouldn't we fail instead of just logging a warning? The user expected the combo to do something and instead the action does nothing. I would be confused.